### ISSUE TITLE
Exception occured When set a Req headers

### ISSUE THREAD

**Issue by sky5454** [RusKnyaz/Optimus#58]
Ver: 2.4.1 
the code is 

it works
```c#
var webProxy = new WebProxy(proxy_new_addr, true);

var engine = EngineBuilder.New()
 .ConfigureResourceProvider(r => r.Http(http => http.Proxy(webProxy))
    .Notify(
        req => { 
            req.Headers["User-Agent"] = header_userAgent;
            //req.Headers["Content-Type"] = header_contentType;
        },
        resp => {
        
        }
 ))
 .UseJint()// Enable JavaScripts execution.
 .Build(); // Builds the Optimus engine


//real_portal_url = ((System.Net.HttpWebResponse)resp).ResponseUri.AbsoluteUri;
await engine.OpenUrl(portal_url);
```
but this code, doesn't work
```c#
var webProxy = new WebProxy(proxy_new_addr, true);

var engine = EngineBuilder.New()
 .ConfigureResourceProvider(r => r.Http(http => http.Proxy(webProxy))
    .Notify(
        req => { 
            //req.Headers["User-Agent"] = header_userAgent;
            req.Headers["Content-Type"] = header_contentType;
        },
        resp => {
        
        }
 ))
 .UseJint()// Enable JavaScripts execution.
 .Build(); // Builds the Optimus engine


//real_portal_url = ((System.Net.HttpWebResponse)resp).ResponseUri.AbsoluteUri;
await engine.OpenUrl(portal_url);
```

the log of 2nd Code here:

```c#
System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (One or more errors occurred. (Object reference not set to an instance of an object.))
  Source=System.Private.CoreLib
  StackTrace:
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Knyaz.Optimus.ResourceProviders.NotifyingResourceProvider.<>c__DisplayClass4_0.<SendRequestAsync>b__0(Task`1 t)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Knyaz.Optimus.Engine.<OpenUrl>d__53.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Knyaz.Optimus.Engine.<OpenUrl>d__52.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at ConsoleApp1.Program.<Main>d__14.MoveNext() in E:\01_NewCode\WPF\ConsoleApp1\Program.cs:line 99

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)

Inner Exception 2:
NullReferenceException: Object reference not set to an instance of an object.
```

**Comment by sky5454** [comment]
I wonder if it cause by thread-safety?

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-5.0#thread-safety

**Comment by RusKnyaz** [comment]
This is because the request contains no content and therefore you can't setup content-type.

The answer can be found here: https://stackoverflow.com/questions/10679214/how-do-you-set-the-content-type-header-for-an-httpclient-request

**Comment by sky5454** [comment]
> This is because the request contains no content and therefore you can't setup content-type.
> 
> Some useful info can be found here: https://stackoverflow.com/questions/10679214/how-do-you-set-the-content-type-header-for-an-httpclient-request

Thanks, it may be better to `new MediaTypeHeaderValue(keyValue.Value);` when `resultRequest.Content` is null
https://github.com/RusKnyaz/Optimus/commit/2ea6c3c4c89b6340de8efed47f30257b76d734dc#commitcomment-49662234

**Comment by sky5454** [comment]
And actually the issue is not to add a Header.
It causes by the problem that `Optimus` send a form without a header
```http
ContentType : "application/x-www-form-urlencoded" 
```
but "application/x-www-form-urlencoded"  is the Form's default value. and it's important.

And see this: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form
![image](https://user-images.githubusercontent.com/12370628/115569527-e3f06c80-a2ef-11eb-81ba-b400f325aae5.png)

**Comment by RusKnyaz** [comment]
I fixed it in 089f2c9b97f33a24bd4a6b74541e62faa17ea764, please check if it works for you.

**Comment by sky5454** [comment]
> I fixed it in [089f2c9](https://github.com/RusKnyaz/Optimus/commit/089f2c9b97f33a24bd4a6b74541e62faa17ea764), please check if it works for you.

Thx, Upgrade to 3.0.0-beta-305 and it works.
